pkg/rbac/common.go

```diff
@@ -0,0 +1,66 @@
+package rbac
+
+import (
+	"github.com/pkg/errors"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	projectv3 "github.com/rancher/types/apis/project.cattle.io/v3"
+	rbacv1 "k8s.io/api/rbac/v1"
+)
+
+/*BuildSubjectFromObject Can accept PRTB, CRTB and APP as input.
+This function will generate PRTB and CRTB to the subject with user or group
+and generate APP to the subject with ServiceAccount
+*/
+func BuildSubjectFromObject(object interface{}) (rbacv1.Subject, error) {
```

Comment:
I'm not sure I like the generic method for everything. I'd rather have one per concrete type. And then have a generic method for subject, where you pass username, groupPrincipalName, groupName, namespace

Reply:
In his defense, he just extend this to includ an App if/else. i had this logic prior